crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,1278 @@
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::exit_codes::ExitCode;
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+use core::f64;
+use indexmap::IndexMap;
+use qiskit_circuit::operations::{Operation, Param, StandardGate};
+use qiskit_circuit::PhysicalQubit;
+use qiskit_transpiler::target::{InstructionProperties, Qargs, Target, TargetError};
+use std::mem::forget;
+use std::ptr::null_mut;
+
+// Helper structures
+
+/// Represent a list of instruction names coming from the target.
+#[repr(C)]
+pub struct QkTargetOpsList {
+    list: *mut StandardGate,
+    length: usize,
+}
+
+/// Represent a list of qargs coming from the target.
+#[repr(C)]
+pub struct QkTargetQargsList {
+    list: *mut QkTargetQargs,
+    length: usize,
+}
+
+/// Represent qargs coming from the target.
+#[repr(C)]
+pub struct QkTargetQargs {
+    args: *mut u32,
+    length: usize,
+}
+
+/// Represents the properties of an instruction in the Target
+#[repr(C)]
+pub struct QkInstructionProps {
+    duration: f64,
+    error: f64,
+}
+
+/// @ingroup QkTarget
+/// Construct a new ``Target`` with the given number of qubits.
+/// The number of qubits is bound to change if an instruction is added with properties
+/// that apply to a collection of qargs in which any index is higher than the specified
+/// number of qubits
+///
+/// @param num_qubits The number of qubits the ``Target`` will explicitly support.
+///
+/// @return A pointer to the new ``Target``
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_target_new(num_qubits: usize) -> *mut Target {
+    let target = Target::new(
+        None,
+        Some(num_qubits),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )
+    .unwrap();
+    Box::into_raw(Box::new(target))
+}
+
+/// @ingroup QkTarget
+/// Returns the number of qubits of this ``Target``.
+///
+/// @param target A pointer to the ``Target``.
+///
+/// @return The number of qubits this target can use.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///     uint32_t num_qubits = qk_target_num_qubits(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_num_qubits(target: *const Target) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.num_qubits.unwrap_or_default()
+}
+
+/// @ingroup QkTarget
+/// Returns the dt value of this ``Target``.
+///
+/// @param target A pointer to the ``Target``.
+///
+/// @return The dt value of this ``Target`` or `NaN` if not assigned.
+///
+/// # Example
+///     const dt = 10e-9;
```

Comment:
It looks like this needs to be removed 🙂 
```suggestion
///
```

Reply:
Removed in 395a194